packages/common/src/client/sync/stream/core-instruction.ts

```diff
@@ -53,3 +69,31 @@ export interface BucketProgress {
 export interface FetchCredentials {
   did_expire: boolean;
 }
+
+function priorityToJs(status: SyncPriorityStatus): sync_status.SyncPriorityStatus {
+  return {
+    priority: status.priority,
+    hasSynced: status.has_synced ?? undefined,
+    lastSyncedAt: status?.last_synced_at != null ? new Date(status!.last_synced_at! * 1000) : undefined
```

Comment:
Nit: I think we can remove some of the operators here?

```suggestion
    lastSyncedAt: status.last_synced_at != null ? new Date(status.last_synced_at! * 1000) : undefined
```